minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -616,15 +630,21 @@ private static void addConnection(final Element parentElement, Map<String, Objec
             addTextElement(element, "labelIndex", "1");
             addTextElement(element, "zIndex", "0");
 
-            addTextElement(element, "sourceId", "Processor");
+            addTextElement(element, "sourceId", getValueString(connectionProperties, SOURCE_NAME_KEY));
```

Comment:
One additional thought I had was to have another class that perhaps does the value checking and enforces the config YAML and then passes a value object (a collection of all the backing maps) to the ConfigTransformer.  Long term, we may not have this particular process and could isolate the actual validation and checking elsewhere.  In lieu of the manual checks, we may be able to make use of something like [Kwalify](http://www.kuwata-lab.com/kwalify/) (no clue as to its quality) which might help in cutting down on the onerous and manual if statements for each individual scalar.